pkg/workloads/spark_job/spark_util.py

```diff
@@ -246,26 +268,27 @@ def ingest(ctx, spark):
 
 def read_csv(ctx, spark):
     data_config = ctx.environment["data"]
-    expected_field_names = data_config["schema"]
-
-    schema_fields = []
-    for field_name in expected_field_names:
-        if field_name in ctx.raw_columns:
-            spark_type = CORTEX_TYPE_TO_SPARK_TYPE[ctx.raw_columns[field_name]["type"]]
-        else:
-            spark_type = StringType()
-
-        schema_fields.append(StructField(name=field_name, dataType=spark_type))
 
     csv_config = {
         util.snake_to_camel(param_name): val
         for param_name, val in data_config.get("csv_config", {}).items()
         if val is not None
     }
 
-    df = spark.read.csv(
-        data_config["path"], schema=StructType(schema_fields), mode="FAILFAST", **csv_config
-    )
+    df = spark.read.csv(data_config["path"], inferSchema=True, **csv_config)
```

Comment:
Is it possible to only infer schema on certain columns? That seems like it would be safer to me (e.g if you have a string column which happens to have values that look like ints). @vishalbollu any thoughts here?

As you're working on this you might also want to take the chance to support underscores for skipping and leaving some of the columns off the end (e.g. if you had columns named 1, 2, 3, 4, 5 and you cared about 1, 2, and 4, you could ingest them with `[1, 2, _, 4]`)

Also, can you add a test that confirms we use the correct types, e.g. something like this:

```python
schema = [a_str, b_float]
csv_str = "1,1"
```

Reply:
It isn't possible to partially infer columns. Better option is to infer all the columns and cast to the known types.